File: 506-relative-ranks.rb
def find_relative_ranks(nums)
    sorted_arr = nums.sort{ |a, b| b <=> a }
    (0...nums.length).each do |i|
        if nums[i] == sorted_arr[0]
            nums[i] = "Gold Medal"
        elsif nums[i] == sorted_arr[1]
            nums[i] = "Silver Medal"
        elsif nums[i] == sorted_arr[2]
            nums[i] = "Bronze Medal"
        else
            nums[i] = (sorted_arr.index(nums[i]) + 1).to_s
        end
    end
    nums
end

File: 506-relative-ranks.js
var findRelativeRanks = function(nums) {
    let newArr = [];
    for (let i = 0; i < nums.length; i++) {
        newArr.push(nums[i]);
    }
    newArr = newArr.sort((a, b) => {return b - a});
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === newArr[0]) {
            nums[i] = "Gold Medal";
        } else if (nums[i] === newArr[1]) {
            nums[i] = "Silver Medal";
        } else if (nums[i] === newArr[2]) {
            nums[i] = "Bronze Medal";
        } else {
            let index = newArr.indexOf(nums[i]);
            nums[i] = (index + 1).toString();
        }
    }

    return nums;
};
